### ISSUE TITLE
Spectrum.livetime null value

### ISSUE THREAD

**Issue by jccurtis** [lbl-anp/becquerel#140]
Currently a `Spectrum` can have numerical (`float`) livetime or two different `null` values if the the `livetime` was not provided. These cases can be found in `Spectrum.__init__`:

1. `None` if `counts` are provided
2. `np.nan` if `cps` are provided

This makes testing for the `livetime` a bit tricky as noted in #129:

> Should this be changed to `np.isnan` like below? Otherwise I find `pd.isnull` catches more `null` cases.
> ```python
> In [11]: pd.isnull(None)
> Out[11]: True
> 
> In [12]: np.isnan(None)
> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> <ipython-input-12-f53008f0bd14> in <module>()
> ----> 1 np.isnan(None)
> 
> TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
> 
> In [13]: pd.isnull(np.nan)
> Out[13]: True
> ```
> That said `np.isnan` is a bit faster for numeric arrays:
> ```python
> In [20]: null_arr = np.array([np.nan, 1.] * 10000)
> 
> In [21]: %timeit np.isnan(null_arr)
> 4.07 µs ± 19 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
> 
> In [22]: %timeit pd.isnull(null_arr)
> 14.6 µs ± 36.2 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
> ```

With this in mind, @markbandstra and I agree that if the `livetime` is not provided (or is no longer relevant - say after a subtraction operation with two different `livetime`'s) then `livetime` should be set to `None` and not the `float` `np.nan`. 

I'll change this and update the tests unless I've missed something in the `core` which depends on this nuance.

**Comment by cosama** [comment]
I gave this matter some thought. I think the reason why there are two different livetimes for counts and cps is that it allows for testing of which fields actually has been set and are available. For example the following code would work:
```
def doSomeThingWithASpectrum(spectrum):
    if spectrum.livetime is None:
        #Do something with spectrum.counts (cps not available)
    elif spectrum.livetime is np.nan:
        #Do something with spectrum.cps (counts not available)
    else:
        #Do something with spectrum.counts or spectrum.cps, livetime is valid
```
If we set livetime always to `None` in the future we would have to do:
cps is that it allows for testing of which fields actually has been set and are available. For example the following code would work:
```
def doSomeThingWithASpectrum(spectrum):
    try:
        #Do something with spectrum.counts (might work or not)
    except SpectrumError:
        #Do something with spectrum.cps (counts not available and livetime not valid)
        pass
```
I think the second version is fine, but there would be no way of telling from now on if the livetime has been set how the spectrum was initiated (through counts, or cps).

**Comment by cosama** [comment]
Actually there is a way to still check how a `Spectrum` was initiated, though the `_counts` and `_cps` members. I think that is sufficient for the rare edge cases were this really is required. I implemented the new behavior in #144
